source/common/stats/thread_local_store.cc

```diff
@@ -87,6 +87,10 @@ void ThreadLocalStoreImpl::releaseScopeCrossThread(ScopeImpl* scope) {
 }
 
 std::string ThreadLocalStoreImpl::getTagsForName(const std::string& name, std::vector<Tag>& tags) {
+  if (default_tags_ != nullptr) {
```

Comment:
Why is this now necessary?

Reply:
Since [we have to initialize stats stores before we ready the `default_tags`](https://github.com/taiki45/envoy/blob/feb9caf44419ed1ebd635b58ec3206a2c7bea003/source/server/server.cc#L196), stats stores have their `default_tags_` fields as a point (`const std::vector<Tag>*`). Pointers can be null, so I thought we need check nullability of the pointer. The `tag_extractors_` also checks the nullability below. Still I'm not sure it's not wrong.

Reply:
Refactored with e0596988a490ae5a0ad78271523d2cc9046d6896 to stop using pointer approach. Instead, use null object pattern to ensure they always exist.